### PR TITLE
Animations

### PR DESCRIPTION
**Pull request by wixysam**

### CODE REVIEW

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 1]
```diff
@@ -0,0 +1,814 @@
...
+              "doc": "Target vertical scale of the animated element or elements.\n\n If a `scale` value is set, also setting a `scaleY` value overrides the vertical\n scaling set use `scale`.",
+              "optional": true },
+            { "name": "x",
+              "type":
+                [ "number",
+                  "string" ],
+              "doc": "Horizontal pixel target to move the element or elements \n to horizontally.\n\n One of the following formats:  \n\n + A positive number to specify a target pixel to the right, relative to the\n   original position.\n + A negative number to specify a target pixel to the left, relative to the\n   original position.\n + A `\"+=\"` expression to specify a target pixel to the right, relative to the\n   last animated position. For example, \"+=100\" moves 100 pixels to the right\n   of wherever the element or elements are positioned when this specific animation\n   begins.\n + A `\"-=\"` expression to specify a target pixel to the left, relative to the\n   last animated position.For example, \"-=100\" moves 100 pixels to the left\n   of wherever the element or elements are positioned when this specific animation\n   begins.",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: Add space before "For Example"

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 2]
```diff
@@ -0,0 +1,814 @@
...
+              "optional": true },
+            { "name": "y",
+              "type":
+                [ "number",
+                  "string" ],
+              "doc": "Vertical pixel target to move the element or elements \n to horizontally.\n\n One of the following formats:\n\n + A positive number to specify a target pixel down, relative to the\n   original position.\n + A negative number to specify a target pixel up, relative to the\n   original position.\n + A `\"+=\"` expression to specify a target pixel down, relative to the\n   last animated position. For example, \"+=100\" moves 100 pixels down\n   from wherever the element or elements are positioned when this specific animation\n   begins.\n + A `\"-=\"` expression to specify a target pixel up, relative to the\n   last animated position.For example, \"-=100\" moves 100 pixels up\n   from wherever the element or elements are positioned when this specific animation\n   begins.",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: vertically not horizontally

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 3]
```diff
@@ -0,0 +1,814 @@
...
+                      "} );" ],
+                  "extra":
+                    {  } } ],
+            "extra":
+              {  } },
+        "extra":
+          {  } } ],
+  "callbacks": [],
+  "messages":
+    [ { "name": "Animation",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: Maybe add another animation example that uses some of the other properties so users can just copy and paste?

File: wixcode-seo/wix-animations.service.json [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  "labels":
+    [ "changed" ],
+  "location":
+    { "lineno": 1,
+      "filename": "animations.js" },
+  "docs":
+    { "summary": "The wix-animations module contains functionality for working with\n animations.",
+      "description":
+        [ "Animation sequences are composed using a timeline. A timeline defines what animations",
+          " are played on which page elements and when those animations are played.",
```
Comment (talyaro) [inline wixcode-seo/wix-animations.service.json]: I think you should explain the typical process - build a timeline, add animations, play. Took me some time to figure that out.

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 5]
```diff
@@ -0,0 +1,814 @@
...
+        {  } },
+  "properties": [],
+  "operations":
+    [ { "name": "add",
+        "labels": [],
+        "nameParams": [],
+        "params":
+          [ { "name": "target",
+              "type":
+                [ "$w.HiddenMixin",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: Why is the parameter a mixin? Does this mean all elements that import the mixin? Very confusing and requires explanation.

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 6]
```diff
@@ -0,0 +1,814 @@
...
+                      "",
+                      "  timeline.play();",
+                      "} );" ],
+                  "extra":
+                    {  } } ],
+            "extra":
+              {  } },
+        "extra":
+          {  } },
+      { "name": "onComplete",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: I guess you decided to make this present tense to match onStart and onRepeat?

File: wixcode-seo/wix-animations/TimeLine.service.json [hunk 7]
```diff
@@ -0,0 +1,814 @@
...
+        "docs":
+          { "summary": "Sets an event handler that runs when a the timeline repeats.",
+            "description":
+              [ "The event handler set by calling the `onRepeat()` function runs when a ",
+                " timeline begins playing a repetition.",
+                "",
+                " If an event handler is already set for `onRepeat`, setting a new event",
+                " handler overrides the one set previously.  ",
+                "",
+                " To remove an event handler you set previously, call the `onComplete()`",
```
Comment (talyaro) [inline wixcode-seo/wix-animations/TimeLine.service.json]: onRepeat()

File: wixcode-seo/wix-animations/TimeLine.service.json (hunk 6 above)
Comment (wixysam) [reply wixcode-seo/wix-animations/TimeLine.service.json]: I didn't decide anything. Never when through spec review. Wasn't interested in making a fuss.

**Pull request closed without merging**